### DISCUSSION TITLE
Text update without blink

### DISCUSSION THREAD

**Discussion by fruktiz32** [Bodmer/TFT_eSPI#1136] [Compatible displays and setup files]
Hello, I want to ask if it is possible to update text without that little blink when you set background black. As I used to work with other display libraries (ssd1306) there you first write functions and then call for display and everything updates instant without little blink.
I want to refresh text from JSON every 5 seconds and don't like that blink everytime. Now my  code looks like this, it refresh every 5s:

>       tft.fillScreen(TFT_BLACK);
>       tft.drawString(String(symbol), 0, 0);
>       tft.drawString("$" + String(price), 0, 20);
>       tft.drawString("24h: " + String(change) + "%", 0, 40);

**Comment by Bodmer** [comment]
The typical SSD1306 library uses a frame buffer in RAM so sequential graphics operations in the buffer, that would cause flicker, are not seen. There is usually not enough memory to use full screen buffers with TFTs, but Sprites can be used for small areas. See examples.

The simplest approach is to not clear the whole screen as this is the main cause of flicker. Clearing the screen takes an appreciable time so is obviously visible. Instead just draw the updated values with the text backgound colour set. If the values do not overwrite completely then use the setTextPadding function, see examples.

**Reply by joey120373** [reply]
In addition to the answer above, what I like to do is :
First, only update the variable to the TFT  when it has changed. To do this you just keep another variable, for example : 
Int x; 
int previous_x; 
After I print x to the TFT, I then do a 
previous_x = x; 

to keep it from flickering and to keep artifacts from Remaining on screen after a new value has been printed, I first print the 
previous_x value using a text color that matches 
The background, then print the new value in whatever color I happen to be using. 

Not sure this is the most effective way of doing things, but keeping track of the previous value of whatever it is you want to print, makes it very easy to only print it if it changes. That alone cuts back on the flickering, since your not writing the same value to the screen repeatedly for no reason.

**Reply by Bodmer** [reply]
This library uses that method and only updates characters that change. It works quite well with numeric values.
https://github.com/KrisKasprzak/FlickerFreePrint
